fhir-server/src/main/java/com/ibm/fhir/server/registry/ServerRegistryResourceProvider.java

```diff
@@ -189,7 +189,7 @@ public void afterDelete(FHIRPersistenceEvent event) {
             MultiResourceResult<Resource> result = persistence.search(context, resourceType);
 
             if (result.isSuccess()) {
-                List<FHIRRegistryResource> registryResources = new ArrayList<>(searchContext.getTotalCount());
+                List<FHIRRegistryResource> registryResources = new ArrayList<>(searchContext.getTotalCount() != null ? searchContext.getTotalCount() : 0);
```

Comment:
This ternary is done in a few places. Perhaps warrants a little helper function?

Reply:
Since it was just a few places, I felt it was the most straightforward option.